src/saltext/github/modules/github.py

```diff
@@ -1855,22 +1861,27 @@ def _query(
             headers=True,
             status=True,
             text=True,
-            hide_fields=["access_token"],
+            hide_fields=header_dict["Authorization"],
             opts=__opts__,
         )
         log.debug("GitHub Response Status Code: %s", result["status"])
 
-        if result["status"] == 200:
+        if result["status"] in [200, 201, 204]:
+            if result["status"] == 204:
+                return result
```

Comment:
**suggestion (non-blocking):** it's probably more readable to ditch the nested `if` statement here

```suggestion
        if result["status"] == 204:
            return result
        if result["status"] in [200, 201]:
```